gprofiler/profilers/dotnet.py

```diff
@@ -0,0 +1,157 @@
+        storage_dir: str,
+        profile_spawned_processes: bool,
+        dotnet_mode: str,
+    ):
+        super().__init__(frequency, duration, stop_event, storage_dir)
+        assert dotnet_mode == "dotnet-trace", \
+                              "Dotnet profiler should not be initialized, wrong dotnet-trace value given"
+        self._metadata = DotnetMetadata(self._stop_event)
+
+    def _make_command(self, process: Process, duration: int) -> List[str]:
+        result = run_process(
+            f"ls /proc/{process.pid}/root/tmp | grep diagnostic ",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            shell=True,
+            suppress_log=True,
+            check=False,
+        )
+        socket_list = result.stdout.decode("utf-8").split('\n')
+        tempdir = f"TMPDIR=/proc/{process.pid}/root/tmp"
+        internal_pid = str(process.pid)
+        metadata = None
+        if int(socket_list[0].split("-")[2]) != 1:
+            # we are not in docker, pid different than 1
+            metadata = self._metadata.get_metadata(process)
+        else:
+            # we are in docker, pid is 1
+            internal_pid = "1"
+        return [
+            "sudo",
+            tempdir,
+            resource_path(self.RESOURCE_PATH),
+            "collect",
+            "--format",
+            "speedscope",
+            "--process-id",
+            internal_pid,
+            "--duration",
+            self._parse_duration(duration),
+            #           str(self._frequency), - TODO: frequency handling to be determined
+        ], metadata
+
+    def _profile_process(self, process: Process, duration: int, spawned: bool) -> ProfileData:
+        logger.info(
+            f"Profiling{' spawned' if spawned else ''} process {process.pid} with dotnet-trace",
+            cmdline=" ".join(process.cmdline()),
+            no_extra_to_server=True,
+        )
+        appid = None
+        app_metadata = None
+        local_output_path = os.path.join(self._storage_dir, f"dotnet-trace.{random_prefix()}.{process.pid}.col")
+        with removed_path(local_output_path):
+            try:
+                command, app_metadata = self._make_command(process, duration)
+                result = run_process(
+                    command,
+                    stop_event=self._stop_event,
+                    timeout=self._duration + self._EXTRA_TIMEOUT,
+                    kill_signal=signal.SIGKILL,
+                )
+                local_output_path = result.stdout.decode("utf-8").split("\t")[1].split("\n")[0]
+            except ProcessStoppedException:
+                raise StopEventSetException
+            logger.info(f"Finished profiling process {process.pid} with dotnet")
+            return ProfileData(load_speedscope_as_collapsed(Path(local_output_path), self._frequency),
+                               appid, app_metadata)
+
+    def _parse_duration(self, duration: int):
```

Comment:
This translates the duration in seconds to HH:MM:ss timestamp, right?

It can be replaced with `str(datetime.timedelta(duration))`, just ensure `duration < 3600 * 24` because then its format changes (more than 1 day). That's not an acceptable usage of gProfiler anyway.